File: packages/backend/src/main/kotlin/hollybike/api/types/invitation/TInvitationCreation.kt
/*
  Hollybike API Kotlin KTor Graalvm application
  Made by MacaronFR (Denis TURBIEZ) and Loïc Vanden Bossche
*/
package hollybike.api.types.invitation

import hollybike.api.types.user.EUserScope
import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class TInvitationCreation(
	val label: String? = null,
	val role: EUserScope,
	val association: Int? = null,
	val maxUses: Int? = null,
	val expiration: Instant? = null
)

File: packages/frontend/src/types/TInvitationCreation.ts
/*
  Hollybike Back-office web application
  Made by MacaronFR (Denis TURBIEZ) and enzoSoa (Enzo SOARES)
*/
export interface TInvitationCreation {
	label: string,
	role: string,
	association?: number,
	maxUses?: number,
	expiration?: string
}
